contrib/babelfishpg_tsql/src/procedures.c

```diff
@@ -3450,6 +3458,119 @@ sp_rename_internal(PG_FUNCTION_ARGS)
+			char *major_name = lowerstr(old_name);
+			char *new_major_name = lowerstr(new_name);
+
+			/* schema_name doesn't need to truncate again. */
+			truncate_tsql_identifier(major_name);
+			truncate_tsql_identifier(new_major_name);
+
+			if (objtype == OBJECT_TABLE)
+			{
+				type = ExtendedPropertyTypeNames[EXTENDED_PROPERTY_TABLE];
+				update_extended_property(db_id, type, schema_name,
+										 major_name, NULL,
+										 Anum_bbf_extended_properties_major_name,
+										 new_major_name);
+				type = ExtendedPropertyTypeNames[EXTENDED_PROPERTY_TABLE_COLUMN];
+				update_extended_property(db_id, type, schema_name,
+										 major_name, NULL,
+										 Anum_bbf_extended_properties_major_name,
+										 new_major_name);
+			}
+			else if (objtype == OBJECT_VIEW)
+			{
+				type = ExtendedPropertyTypeNames[EXTENDED_PROPERTY_VIEW];
+				update_extended_property(db_id, type, schema_name,
+										 major_name, NULL,
+										 Anum_bbf_extended_properties_major_name,
+										 new_major_name);
+			}
+			else if (objtype == OBJECT_SEQUENCE)
+			{
+				type = ExtendedPropertyTypeNames[EXTENDED_PROPERTY_SEQUENCE];
+				update_extended_property(db_id, type, schema_name,
+										 major_name, NULL,
+										 Anum_bbf_extended_properties_major_name,
+										 new_major_name);
+			}
+			else if (objtype == OBJECT_PROCEDURE)
+			{
+				type = ExtendedPropertyTypeNames[EXTENDED_PROPERTY_PROCEDURE];
+				update_extended_property(db_id, type, schema_name,
+										 major_name, NULL,
+										 Anum_bbf_extended_properties_major_name,
+										 new_major_name);
+			}
+			else if (objtype == OBJECT_FUNCTION)
+			{
+				type = ExtendedPropertyTypeNames[EXTENDED_PROPERTY_FUNCTION];
+				update_extended_property(db_id, type, schema_name,
+										 major_name, NULL,
+										 Anum_bbf_extended_properties_major_name,
+										 new_major_name);
+			}
+			else if (objtype == OBJECT_TYPE)
+			{
+				type = ExtendedPropertyTypeNames[EXTENDED_PROPERTY_TYPE];
+				update_extended_property(db_id, type, schema_name,
+										 major_name, NULL,
+										 Anum_bbf_extended_properties_major_name,
+										 new_major_name);
+			}
+		}
+	}
+	else if (objtype == OBJECT_COLUMN)
+	{
+		if (var_schema_name && var_major_name && old_name)
+		{
+			char *schema_name = get_physical_schema_name(db_name,
+														 lowerstr(var_schema_name));
+			char *major_name = lowerstr(var_major_name);
+			char *minor_name = lowerstr(old_name);
+			char *new_minor_name = lowerstr(new_name);
+
+			/* schema_name doesn't need to truncate again. */
+			truncate_tsql_identifier(major_name);
+			truncate_tsql_identifier(minor_name);
+			truncate_tsql_identifier(new_minor_name);
+
+			type = ExtendedPropertyTypeNames[EXTENDED_PROPERTY_TABLE_COLUMN];
+			update_extended_property(db_id, type, schema_name,
+									 major_name, minor_name,
+									 Anum_bbf_extended_properties_minor_name,
+									 new_minor_name);
+		}
+	}
```

Comment:
must catch the case that objtype is none of the object types tested for

Reply:
If objtype is none of the  object types tested for, ProcessUtility will fail and ereport error, rename_extended_property won't be executed.